webapp/app/js/services/jobfilters.js

```diff
@@ -471,7 +484,7 @@ treeherder.factory('thJobFilters', [
             }
             result = platform + " " + job.platform_option;
         }
-        return result;
+        return String(result);
```

Comment:
Before we were only filtering on fields that were strings in the ``job`` object already.  But with ``tier`` it's an int.  So, I need to cast the ``job.tier`` field value to a string for compare.